contracts/AssetPool.sol

```diff
@@ -1,41 +1,56 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.4.22 <0.9.0;
+pragma solidity ^0.8.0;
 import "./LiquidityPoolInterfaces.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "hardhat/console.sol";
+import "./Proxy.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 
-contract AssetPool is LiquidityPoolInterfaces, Ownable, ReentrancyGuard, ERC20("Cruize Protection LP Token", "cr") {
+contract AssetPool is Proxy, Ownable, ReentrancyGuard {
   using SafeMath for uint256;
-  mapping(address => mapping(address => uint)) public userInfo;
+  mapping(address => mapping(address => uint256)) public userInfo;
   mapping(address => bool) private allowedAssets;
+  mapping(address => address) public assetToLp;
+  address[] public tokenAddress;
+
+
+  event Withdraw(
+      address indexed account,
+      uint256 writeAmount
+  );
+  event Provide(address indexed account,uint256 writeAmount);
+
+  // This is proxy code
+  address immutable masterContract;
+
+  constructor(address _masterContract) {
+    masterContract = _masterContract;
+  }
+
+  function createToken(string memory name_, string memory symbol_, address assetAddress) external onlyOwner nonReentrant {
+    CruizeToken token = CruizeToken(createClone(masterContract));
+    token.init(name_, symbol_);
+    assetToLp[assetAddress] = address(token);
+  }
+  // Proxy code ends here
 
   function depositAsset(uint256 depositAmount, address assetAddress) external payable nonReentrant {
```

Comment:
asset address can be changed to reserve  , and deposit  amount could be change to amount

Reply:
why this function is payable, as we are not handling ethers in this smart contract so if someone unintentionally
send the ethers to this function then those ethers will be locked forever